tests/e2e/cluster_downgrade_test.go

```diff
@@ -235,3 +242,22 @@ func getMemberVersionByCurl(cfg *e2e.EtcdProcessClusterConfig, member e2e.EtcdPr
 	}
 	return result, nil
 }
+
+func getVersionFromBinary(binaryPath string) (*semver.Version, error) {
+	lines, err := e2e.RunUtilCompletion([]string{binaryPath, "--version"}, nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not find binary version from %s, err: %w", binaryPath, err)
+	}
+
+	for _, line := range lines {
+		if strings.HasPrefix(line, "etcd Version:") {
+			versionString := strings.TrimSpace(strings.SplitAfter(line, ":")[1])
+			if strings.HasSuffix(versionString, "-alpha.0") {
+				versionString = strings.Replace(versionString, "-alpha.0", "", 1)
+			}
```

Comment:
```suggestion
```

Reply:
true, I try to remember why I added this last week. I thought this specific version wouldn't parse correctly. I'll try this pre-release trick again tomorrow.

Reply:
works, using that instead.